app/shared/session/useLoginErrors.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect } from "react";
+import { useSearchParams } from "react-router-dom";
+
+export const LOGIN_ERROR_CODES = {
+  LOGIN_DEV: "login_dev",
+  LOGIN_GITHUB: "login_github",
+  LOGIN_GOOGLE: "login_google",
+};
+
+export const LOGIN_ERROR_MESSAGES = {
+  [LOGIN_ERROR_CODES.LOGIN_DEV]:
+    "There has been an issue logging you in with dev",
+  [LOGIN_ERROR_CODES.LOGIN_GITHUB]:
+    "There has been an issue logging you in with Github",
+  [LOGIN_ERROR_CODES.LOGIN_GOOGLE]:
+    "There has been an issue logging you in with Google",
+};
+
+export const useLoginErrors = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const error = searchParams.get("error");
+  const message = searchParams.get("message");
+  useEffect(() => {
+    if (error !== null && !!message) {
```

Comment:
please don't do double negation thing, its always harder to read and less explicit in this case than Boolean(message) or message !== ''

Reply:
updatedc to boolean